### PR TITLE
Fixes #17574: clickhouse lineage processing

### LINKED ISSUE #17574: No relations for clickhouse views (and materialized) in lineage
**Affected module**
Lineage view for Clickhouse

**Describe the bug**
No relations for clickhouse views (and materialized views) in lineage

**To Reproduce**
1) Ingest metadata from clickhouse.
2) choose view (or materialized view) in the list of ch table
3) click lineage tab

Screenshots or steps to reproduce
mv_ch

**Expected behavior**
I expect the UI to show me the relation from the materialized view schema definition. 

**Version:**
 - OS: macOS Ventura 13.5
 - Python version: 3.11.7
 - OpenMetadata version: 1.4.8
 - OpenMetadata Ingestion package version: openmetadata-ingestion[docker]==1.4.8.

**Additional context**
Docker version 25.0.2
Docker Compose version v2.24.3-desktop.1

### PR DESCRIPTION
**Pull request by verejoel**
### Describe your changes:

(disclaimer: problem was diagnosed and solution coded with Claude 4)

Fixes #17574

Fixed Clickhouse lineage extraction failing to resolve schema-prefixed table names (e.g., `schema.table_name`). Clickhouse queries commonly use this format, but the generic lineage logic couldn't handle the database-as-schema architecture differences.

**Changes:**
- Added multi-strategy table resolution in `ClickhouseLineageSource.get_clickhouse_table_entities_from_query()`
- Implemented 5 fallback strategies: original name, parsed schema as database_schema, parsed schema as database_name (Clickhouse-specific), base table name, and case variations
- Enhanced logging to show failed resolution attempts for debugging

**Testing:**
- Added comprehensive test suite with 14 test cases covering all strategies, edge cases, and error scenarios
- Tests use real `ClickhouseLineageSource` implementation with proper mocking
- All tests pass, validating the fix works correctly

### Type of change:
- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.
- [x] I have added a test that covers the exact scenario we are fixing. For complex issues, comment the issue number in the test for future reference.

### CODE REVIEW

**Comment by harshach** [comment]
@verejoel thanks for the PR. @ulixius9 please review

File: ingestion/src/metadata/ingestion/source/database/clickhouse/lineage.py [hunk 1]
```diff
@@ -36,5 +47,134 @@ class ClickhouseLineageSource(ClickhouseQueryParserSource, LineageSource):
...
+                [
+                    # Try parsed schema as database_schema
+                    (database_name, schema_part, table_part, "parsed_schema_as_schema"),
+                    # Try parsed schema as database_name (Clickhouse databases-as-schemas)
+                    (
+                        schema_part,
+                        database_schema,
+                        table_part,
+                        "parsed_schema_as_database",
+                    ),
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/database/clickhouse/lineage.py:95]: this situation should never occur right because clickhouse is a single database source and in OpenMetadata we always mark database as `default` by default so schema name can never be at the place of database 🤔

Comment (verejoel) [reply ingestion/src/metadata/ingestion/source/database/clickhouse/lineage.py:95]: I agree. I will remove this strategy.

**Comment by ulixius9** [comment]
this should be already fixed with latest release. feel free to reopen the issue if still facing problem
